### PR TITLE
[BREAKINGCHANGE] TimeSeriesValueTuple null supported, `connect_nulls` option added

### PR DESCRIPTION
**Pull request by sjcobb**
Updates to allow time series queries to return null values. Summary for changelog:

- [BREAKINGCHANGE] TimeSeriesValueTuple value now accepts null
- [FEATURE] Support connect_nulls option in time series panel
- [BUGFIX] Do not show nulls as zero in tooltip

### Recording (new panel option)

https://user-images.githubusercontent.com/9369625/221610277-e1eadd61-ba1e-467c-8811-06217b1ec61b.mp4

### Screenshots
`connect_nulls` false ex:
connect_nulls_false_ex

`connect_nulls` true ex:
connect_nulls_true_ex png

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).
- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.

### CODE REVIEW

**Review by saminzadeh — APPROVED** [review]
Great thanks for doing this! I guess have the points on the graphs will be more important now if the lines are not connected:

image

File: ui/plugin-system/src/model/calculations.ts [hunk 1]
```diff
@@ -75,5 +76,9 @@ function sum(values: TimeSeriesValueTuple[]): number | undefined {
 }
 
 function getValue(valueTuple: TimeSeriesValueTuple) {
-  return valueTuple[1];
+  const value = valueTuple[1];
+  if (value !== null) {
+    return value;
+  }
+  return NaN;
```
[the remark this answers is not in the document]
Comment (sjcobb) [reply ui/plugin-system/src/model/calculations.ts:83]: I tried that at first and it breaks some of the existing utils, this was the most low risk way to make the change without further refactoring. Maybe a follow up PR where we refactor them, add more unit tests, and support more options like `LastNotNull` could make sense?

**Comment by sjcobb** [comment]
> points on the graphs will be more important now if the lines are not connected

Yeah that makes sense, as we discussed offline, I'm going to add a panel option to force points to show

File: ui/components/src/StatChart/StatChart.tsx [hunk 2]
```diff
@@ -24,23 +23,14 @@ import { formatValue, UnitOptions } from '../model/units';
...
-]);
+use([EChartsLineChart, GridComponent, DatasetComponent, TitleComponent, TooltipComponent, CanvasRenderer]);
 
 const MIN_VALUE_SIZE = 12;
 const MAX_VALUE_SIZE = 36;
 
 export interface StatChartData {
   calculatedValue?: number;
   seriesData?: GraphSeries;
-  name?: string;
```
Comment (sjcobb) [inline ui/components/src/StatChart/StatChart.tsx:43]: @eunicorn I cleaned up a few types that weren't used anymore in this chart while I was making adjustments to support `null`, sorry for any merge conflicts with your thresholds branch!

**Pull request merged**